photon-core/src/test/java/org/photonvision/vision/target/TargetCalculationsTest.java

```diff
@@ -49,43 +57,104 @@ public class TargetCalculationsTest {
+                new Point(imageCenterPoint.x + targetPixelOffsetX, imageCenterPoint.y + targetPixelOffsetY);
+
+        var targetYawPitch =
+                TargetCalculations.calculateYawPitch(
+                        imageCenterPoint.x,
+                        targetCenterPoint.x,
+                        params.horizontalFocalLength,
+                        imageCenterPoint.y,
+                        targetCenterPoint.y,
+                        params.verticalFocalLength);
+
+        assertTrue(targetYawPitch.getFirst() > 0, "Yaw is not positive right");
+        assertTrue(targetYawPitch.getSecond() < 0, "Pitch is not positive up");
+
+        var fovs =
+                FrameStaticProperties.calculateHorizontalVerticalFoV(
+                        diagFOV, (int) imageSize.width, (int) imageSize.height);
+        var maxYaw =
+                TargetCalculations.calculateYawPitch(
+                        imageCenterPoint.x,
+                        2 * imageCenterPoint.x,
+                        params.horizontalFocalLength,
+                        imageCenterPoint.y,
+                        imageCenterPoint.y,
+                        params.verticalFocalLength);
+        assertEquals(fovs.getFirst() / 2.0, maxYaw.getFirst(), 0.025, "Horizontal FOV check failed");
+        var maxPitch =
+                TargetCalculations.calculateYawPitch(
+                        imageCenterPoint.x,
+                        imageCenterPoint.x,
+                        params.horizontalFocalLength,
+                        imageCenterPoint.y,
+                        0,
+                        params.verticalFocalLength);
+        assertEquals(fovs.getSecond() / 2.0, maxPitch.getSecond(), 0.025, "Vertical FOV check failed");
     }
 
-    @Test
-    public void pitchTest() {
-        var targetPixelOffsetY = 100;
-        var targetCenterPoint = new Point(imageCenterPoint.x, imageCenterPoint.y + targetPixelOffsetY);
-
-        var truePitch =
-                Math.atan((imageCenterPoint.y - targetCenterPoint.y) / params.verticalFocalLength);
-
-        var pitch =
-                TargetCalculations.calculatePitch(
-                        imageCenterPoint.y, targetCenterPoint.y, params.verticalFocalLength);
+    private static Stream<Arguments> testYawPitchCalcArgs() {
+        return Stream.of(
+                // (yaw, pitch) in degrees
+                Arguments.of(0, 0),
+                Arguments.of(10, 0),
+                Arguments.of(0, 10),
+                Arguments.of(10, 10),
+                Arguments.of(-10, -10),
+                Arguments.of(30, 45),
+                Arguments.of(-45, -20));
+    }
 
-        assertEquals(Math.toDegrees(truePitch) * -1, pitch, 0.025, "Pitch not as expected");
+    private static double[] testCameraMatrix = {240, 0, 320, 0, 240, 320, 0, 0, 1};
+
+    @ParameterizedTest
+    @MethodSource("testYawPitchCalcArgs")
+    public void testYawPitchCalc(double yawDeg, double pitchDeg) {
```

Comment:
Added a test verifying the calculated target yaw/pitch. This fails with the previous calculation, but passes with this PR.